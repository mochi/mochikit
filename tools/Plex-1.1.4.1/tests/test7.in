aa bbb bb bbbbb b ccc
a bb bbb b cc
bb bbb bbbbb  bb bbb
ddddddddddddddd
aaa bbb bb cccc


