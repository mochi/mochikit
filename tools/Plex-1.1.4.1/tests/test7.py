import Test
import sys
from Plex import *

spaces = Rep1(Any(" \t\n"))

lex = Lexicon([
  (Bol + Rep1(Str("a")),       'begin'),
  (      Rep1(Str("b")),       'middle'),
  (      Rep1(Str("c")) + Eol, 'end'),
  (Bol + Rep1(Str("d")) + Eol, 'everything'),
  (spaces, IGNORE)
],
debug = Test.debug,
timings = sys.stderr
)

Test.run(lex, "test7", trace = 0)


