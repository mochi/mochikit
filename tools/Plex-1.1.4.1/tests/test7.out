test7.in   1   0 begin      'aa'
test7.in   1   3 middle     'bbb'
test7.in   1   7 middle     'bb'
test7.in   1  10 middle     'bbbbb'
test7.in   1  16 middle     'b'
test7.in   1  18 end        'ccc'
test7.in   2   0 begin      'a'
test7.in   2   2 middle     'bb'
test7.in   2   5 middle     'bbb'
test7.in   2   9 middle     'b'
test7.in   2  11 end        'cc'
test7.in   3   0 middle     'bb'
test7.in   3   3 middle     'bbb'
test7.in   3   7 middle     'bbbbb'
test7.in   3  14 middle     'bb'
test7.in   3  17 middle     'bbb'
test7.in   4   0 everything 'ddddddddddddddd'
test7.in   5   0 begin      'aaa'
test7.in   5   4 middle     'bbb'
test7.in   5   8 middle     'bb'
test7.in   5  11 end        'cccc'
test7.in   8   0 None       ''
